IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/Browser.cs

```diff
@@ -35,106 +43,52 @@ public Browser(BrowserType browser)
-
-            driver.Manage().Window.Maximize();
-            driver.Manage().Timeouts().PageLoad = TimeSpan.FromMinutes(2);
-            return driver;
-        }
-
-        // Figure out a good way to allow a global override on the wait timeout.
-
         /// <summary>
         /// Wraps the Selenium Driver's native web element to wait until the element exists before returning.
-        /// If you need to verify an element DOESN'T exist, then call Browser.Driver.FindElement directly.
         /// </summary>
         /// <param name="by">Selenium "By" statement to find the element</param>
         /// <param name="secondsToWait">Seconds to wait while retrying before failing</param>
         /// <returns></returns>
-        public Task<IWebElement> FindElement(By by, int secondsToWait = 15)
+        public async Task<IWebElement> FindElement(By by, int secondsToWait = 15)
         {
+            // Figure out a good way to allow a global override on the wait timeout.
             ConditionalWait wait = new ConditionalWait();
-            return wait.WaitFor<NoSuchElementException, StaleElementReferenceException, IWebElement>(() => Driver.FindElement(by), TimeSpan.FromSeconds(secondsToWait));
+            return await wait.WaitFor<NoSuchElementException, StaleElementReferenceException, IWebElement>(() => Driver.FindElement(by), TimeSpan.FromSeconds(secondsToWait));
         }
 
         /// <summary>
         /// Wraps the Selenium Driver's native web element to wait until at least one element exists before returning.
-        /// If you need to verify an element DOESN'T exist, then call Browser.Driver.FindElements directly.
         /// </summary>
         /// <param name="by">Selenium "By" statement to find the element</param>
         /// <param name="secondsToWait">Seconds to wait while retrying before failing</param>
         /// <returns></returns>
-        public Task<IReadOnlyCollection<IWebElement>> FindElements(By by, int secondsToWait = 15)
+        public async Task<IReadOnlyCollection<IWebElement>> FindElements(By by, int secondsToWait = 15)
         {
-            // NOTE: Per conversation with Yuriy on Thursday, this should return an empty collection if nothing is found.
-            // Often times the use case is to assert on the collection, even if nothing is there.
             ConditionalWait wait = new ConditionalWait();
             try
             {
-                return wait.WaitFor<NoSuchElementException, StaleElementReferenceException, IReadOnlyCollection<IWebElement>>(() => Driver.FindElements(by), TimeSpan.FromSeconds(secondsToWait));
+                return await wait.WaitFor<NoSuchElementException, StaleElementReferenceException, IReadOnlyCollection<IWebElement>>(() => Driver.FindElements(by), TimeSpan.FromSeconds(secondsToWait));
             }
             catch(AggregateException)
             {
-                return null;
+                return new ReadOnlyCollection<IWebElement>(new List<IWebElement>());
```

Comment:
A slightly better option here would be to return `Array.Empty<IWebElement>()`. With what you have now it will allocate two objects (the list and the read only collection). Array.Empty will only ever do a single allocation (per type) during the life of your application.